catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java

```diff
@@ -32,110 +30,115 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
 import java.util.stream.Stream;
-import javax.annotation.Nullable;
 import org.apache.commons.lang3.StringUtils;
-import org.apache.http.NameValuePair;
 import org.apache.http.client.utils.URLEncodedUtils;
 import org.codice.ddf.configuration.SystemBaseUrl;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class ImagingChipActionProvider implements MultiActionProvider {
 
-  public static final String TITLE = "Chip Image";
+  static final String TITLE = "Chip Image";
 
-  public static final String DESCRIPTION =
+  static final String DESCRIPTION =
       "Opens a new window to enter the boundaries of an image chip for a Metacard.";
 
-  public static final String PATH = "/chipping/chipping.html";
+  private static final String PATH = "/chipping/chipping.html";
 
-  public static final String ID = "catalog.data.metacard.image.chipping";
+  static final String ID = "catalog.data.metacard.image.chipping";
 
   private static final Logger LOGGER = LoggerFactory.getLogger(ImagingChipActionProvider.class);
 
-  private static final String NITF_IMAGE_METACARD_TYPE = "isr.image";
+  static final String NITF_IMAGE_METACARD_TYPE = "isr.image";
 
-  private static final String ORIGINAL_QUALIFIER = "original";
+  private static final Pattern ALLIANCE_DOWNLOAD_RESOURCE_PATH_PATTERN =
+      Pattern.compile("/services/catalog/sources/.*/.*");
 
-  private static final String QUALIFIER_KEY = "qualifier";
-
-  private GeometryFactory geometryFactory = new GeometryFactory();
+  private final GeometryFactory geometryFactory = new GeometryFactory();
 
   @Override
   public <T> List<Action> getActions(T input) {
-    if (input == null) {
-      LOGGER.debug("Metacard can not be null.");
-      return new ArrayList<>();
-    }
-
-    if (canHandle(input)) {
```

Comment:
`canHandle` is always called before `getActions`. I removed the null and `canHandle` checks here and the corresponding `getAction` unit tests.